app/scripts/modules/core/src/plugins/deck.plugin.ts

```diff
@@ -25,10 +27,14 @@ export interface IDeckPlugin {
 export function registerPluginExtensions(plugin: IDeckPlugin): PromiseLike<any> {
   plugin.stages?.forEach((stage) => Registry.pipeline.registerStage(stage));
   plugin.preconfiguredJobStages?.forEach((stage) => Registry.pipeline.registerPreconfiguredJobStage(stage));
-  plugin.resourceKinds?.forEach((kind) => resourceManager.registerResource(kind));
   toPairs(plugin.help ?? {}).forEach(([key, value]) => HelpContentsRegistry.register(key, value));
   plugin.search?.forEach((search) => searchResultTypeRegistry.register(search));
 
+  const managedDeliveryPlugin: IManagedDeliveryPlugin = {
+    ...plugin.managedDelivery,
+    resources: plugin.resourceKinds || plugin.managedDelivery?.resources,
+  };
+  registerPlugin(managedDeliveryPlugin);
```

Comment:
```suggestion
  if (plugin.managedDelivery || plugin.resourceKinds) {
    const managedDeliveryPlugin: IManagedDeliveryPlugin = {
      ...plugin.managedDelivery,
      resources: plugin.resourceKinds || plugin.managedDelivery?.resources,
    };
    registerPlugin(managedDeliveryPlugin);
  }
```

Reply:
Why do we need this check?

Reply:
most plugins won't have this property, and my opinion is that it would improve clarity.  As a reader of this code, I don't know if registerPlugin does something or nothing when my plugin doesn't export a MD plugin.